src/stream/src/task/actor_manager.rs

```diff
@@ -202,6 +210,157 @@ impl StreamActorManager {
         }
     }
 
+    #[expect(clippy::too_many_arguments)]
+    async fn create_sink_into_table_union(
+        &self,
+        fragment_id: FragmentId,
+        union_node: &stream_plan::StreamNode,
+        env: StreamEnvironment,
+        store: impl StateStore,
+        actor_context: &ActorContextRef,
+        vnode_bitmap: Option<Bitmap>,
+        has_stateful: bool,
+        subtasks: &mut Vec<SubtaskHandle>,
+        local_barrier_manager: &LocalBarrierManager,
+        prefix_nodes: Vec<&stream_plan::StreamNode>,
+        merge_projects: Vec<(&stream_plan::StreamNode, &stream_plan::StreamNode)>,
+    ) -> StreamResult<Executor> {
+        let mut input = Vec::with_capacity(union_node.get_input().len());
+
+        for input_stream_node in prefix_nodes {
+            input.push(
+                self.create_nodes_inner(
+                    fragment_id,
+                    input_stream_node,
+                    env.clone(),
+                    store.clone(),
+                    actor_context,
+                    vnode_bitmap.clone(),
+                    has_stateful,
+                    subtasks,
+                    local_barrier_manager,
+                )
+                .await?,
+            );
+        }
+
+        // Use the first MergeNode to fill in the info of the new node.
```

Comment:
Why do we use `MergeNode` to fill in the info of the `UpstreamSinkUnion` node? Looks really weird to me.